apps/challenges/aws_utils.py

```diff
@@ -689,17 +689,53 @@ def restart_workers_signal_callback(sender, instance, field_name, **kwargs):
     prev = getattr(instance, "_original_{}".format(field_name))
     curr = getattr(instance, "{}".format(field_name))
     if prev != curr:
+        challenge = None
         if field_name == "test_annotation":
             challenge = instance.challenge
         else:
             challenge = instance
-        restart_workers([challenge])
+
+        response = restart_workers([challenge])
+        count, failures = response["count"], response["failures"]
+
         logger.info(
             "The worker service for challenge {} was restarted, as {} was changed.".format(
-                instance.pk, field_name
+                challenge.pk, field_name
             )
         )
 
+        if (count != 1):
+            logger.warning("Worker(s) for challenge {} couldn't restart! Error: {}".format(challenge.id, failures[0]["message"]))
+        else:
+            challenge_url = "https://{}/web/challenges/challenge-page/{}".format(settings.HOSTNAME, challenge.id)
+            challenge_manage_url = "https://{}/web/challenges/challenge-page/{}/manage".format(settings.HOSTNAME, challenge.id)
+
+            if field_name is "test_annotation":
+                file_updated = "Test Annotation"
+            else if field_name is "evaluation_script":
```

Comment:
is there a case for multiple field updates simultaneously? do we need to handle that separately?

Reply:
I mentioned this in the channel as well, Rishabh asked to do this later.